src/Common/src/CoreLib/System/Text/ValueStringBuilder.cs

```diff
@@ -24,10 +24,14 @@ public ValueStringBuilder(Span<char> initialBuffer)
         }
 
         public ValueStringBuilder(int initialCapacity)
+            : this(initialCapacity, false)
+        { }
+
+        public ValueStringBuilder(int initialCapacity, bool appendDefault)
         {
             _arrayToReturnToPool = ArrayPool<char>.Shared.Rent(initialCapacity);
             _chars = _arrayToReturnToPool;
-            _pos = 0;
+            _pos = appendDefault ? _chars.Length : 0;
```

Comment:
Please revert the above changes.  They will not only add cost to existing use of ValueStringBuilder, they are also incorrect: nothing guarantees that the space between indices 0 and _chars.Length are default values, and also it's very likely that _chars.Length > initialCapacity, which means _pos will be as well.

If you want to append default characters, you can do it separately from construction.

Reply:
Could you please explain why it's not guaranteed to have a char array filled with the default values? Do you mean that ArrayPool doesn't reset its state? When you create a new array instance via "new" operator, it's guaranteed to have all items initialized to default values.

Reply:
> Do you mean that ArrayPool doesn't reset its state?

Correct